common/lib/client/auth.js

```diff
@@ -494,6 +501,12 @@ var Auth = (function() {
 	 */
 	Auth.prototype.getAuthHeaders = function(callback) {
 		if(this.method == 'basic') {
+			if (!this.rest.options.tls) {
+				var msg = 'invalid use of Basic auth over non-TLS transport';
+				Logger.logAction(Logger.LOG_ERROR, 'Rest.getAuthHeaders()', msg );
```

Comment:
I believe these are probably `LOG_FATAL` as it's a terminal condition i.e. the library can no longer be used. @paddybyers is that right?